### DISCUSSION TITLE
Dynamic ORDER BY

### DISCUSSION THREAD

**Discussion by rushikeshg25** [prisma/prisma#24703] [Q&A] [answered]
I have a variable sortBy which is either title or createdAt. I want my data to be sorted based of title when sortBy is title and based of creation time when sortBy is createdAt. I tried searching on how to do this but was unsuccessfull

## Alternatives
For now the workaround is having two queries with a if else block like:
```
if (sortBy === "title") {
    data = await prisma.data.findMany({
      where: {
        userId: user.id,
      },
      orderBy: {
        user: {
          title: "asc",
        },
      },
    });
  } else if(sortBy === "createdAt"){
    data = await prisma.data.findMany({
      where: {
        userId: user.id,
      },
      orderBy: {
        user: {
          createdAt: "desc",
        },
      },
    });
  }
```
### Expected
Somewhat like this
```
 data = await prisma.data.findMany({
      where: {
        userId: user.id,
      },
      orderBy: {
        sortBy==="createdAt"?{createdAt:"desc"}:{ title:"desc" }
        },
      },
    });
```
Really sorry if there is already a way to do this. Thank You!

**Comment by Abdel-Monaam-Aouini** [comment, marked as answer]
`orderBy: {
  user: {
    ...( sortBy==="createdAt"?{createdAt:"desc"}:{ title:"desc" })
  }
}`
